apps/web/src/pages/settings/user-profile-page/UserProfileUpdatePasswordForm.tsx

```diff
@@ -0,0 +1,145 @@
+import { showNotification } from '@mantine/notifications';
+import { Button, IconOutlineLockPerson, PasswordInput } from '@novu/design-system';
+import { checkIsResponseError, IResponseError } from '@novu/shared';
+import { api } from '@novu/shared-web';
+import * as Sentry from '@sentry/react';
+import { useMutation } from '@tanstack/react-query';
+import { useForm } from 'react-hook-form';
+import { css } from '../../../styled-system/css';
+import { Stack } from '../../../styled-system/jsx';
+import { PasswordRequirementPopover } from '../../auth/components/PasswordRequirementPopover';
+import { SHARED_PASSWORD_INPUT_REGISTER_OPTIONS } from './UserProfilePasswordSidebar.shared';
+
+type UserProfileUpdatePasswordFormProps = {
+  onSuccess?: () => void;
+};
+
+interface IPasswordUpdateData {
+  currentPassword: string;
+  newPassword: string;
+  confirmPassword: string;
+}
+
+export const UserProfileUpdatePasswordForm: React.FC<UserProfileUpdatePasswordFormProps> = ({ onSuccess }) => {
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** Given that this form and the other password form have different field names, different numbers of inputs, and hit different APIs, it didn't seem worthwhile to try to make an abstraction to reuse code -- especially since they might diverge more in the future.‏

Reply:
Since we are using TypeScript we can start replacing React.FC with typed functions ([more info](https://github.com/typescript-cheatsheets/react#reacttypescript-cheatsheets:~:text=Why%20is%20React.FC%20not%20needed%3F))

Reply:
As you said with TS 5.1 and React 18 it's mostly a matter of preference. My argument aligns with [this post](https://www.totaltypescript.com/you-can-stop-hating-react-fc#should-you-use-it) and is basically; it just reads simpler to do:

`export const UserProfileUpdatePasswordForm = ({ onSuccess }: UserProfileUpdatePasswordFormProps) => { ... }`

Reply:
I generally agree with Matt Pocock and enjoy his content in general, but in this case I feel that the typing is less clear and resembles a normal function (which is not necessarily what I want since I think Components should be considered "first-class" constructs in React).

Either way, we do not have a code styling guide yet, and this is something that we can decide on as a group and put into the guide & accompanying linting